scripts/novaseq/stage/demux-novaseq.bash

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+set -eu
+
+ulimit -n 4096
+
+##########
+# PARAMS #
+##########
+
+IN_DIR=${1?'please provide a run dir'}
+DEMUXES_DIR=${2?'please provide the demuxes dir'}
+FC=${3?'fc_id needed'}
+PROJECTLOG=${4?'projectlog needed'}
+
+RUN=$(basename ${IN_DIR})
+OUT_DIR=${DEMUXES_DIR}/${RUN}
+#SCRIPT_DIR=/home/proj/${CONDA_DEFAULT_ENV}/bin/git/demultiplexing/scripts/novaseq/  # use this when developing in a conda env
+SCRIPT_DIR=/home/proj/${ENVIRONMENT}/bin/git/demultiplexing/scripts/novaseq/        # use this when testing on stage
+#EMAIL=clinical-demux@scilifelab.se
+EMAIL=your.name@scilifelab.se
+
+SLURM_ACCOUNT=development
+if [[ ${ENVIRONMENT} == 'production' ]]; then
+    SLURM_ACCOUNT=production
+fi
+
+#############
+# FUNCTIONS #
+#############
+
+log() {
+    local NOW=$(date +"%Y%m%d%H%M%S")
+    echo "[$NOW] $@"
+}
+
+########
+# MAIN #
+########
+
+# init
+mkdir -p ${OUT_DIR}
+
+# Here we go!
+log "Starting NovaSeq demultiplexing"
+# Send a mail that demultiplexing has started
+cat ${PROJECTLOG} | mail -s "Starting demultiplexing of novaseq flowcell ${FC} on $(hostname)" $EMAIL
+
+BASEMASK=$(demux basemask create --application nova ${IN_DIR})
+UNALIGNED_DIR=Unaligned-${BASEMASK//,}
+
+# DEMUX !
+JOB_TITLE=Demux_${RUN}
+log "sbatch --wait -A ${SLURM_ACCOUNT} -J ${JOB_TITLE} -o ${PROJECTLOG} ${SCRIPT_DIR}/demux-novaseq.batch ${IN_DIR} ${OUT_DIR} ${BASEMASK} ${UNALIGNED_DIR}"
+RES=$(sbatch --wait -A ${SLURM_ACCOUNT} -J ${JOB_TITLE} -o ${PROJECTLOG} ${SCRIPT_DIR}/demux-novaseq.batch ${IN_DIR} ${OUT_DIR} ${BASEMASK} ${UNALIGNED_DIR})
+
+log "bcl2fastq finished!"
+
+# Add samplesheet to unaligned folder
+cp ${IN_DIR}/SampleSheet.csv ${OUT_DIR}/${UNALIGNED_DIR}/
+
+# Restructure the output dir!
+FC=${RUN##*_}
+FC=${FC:1}
+shopt -s nullglob
```

Comment:
```
nullglob
    If set, bash allows patterns which match no files (see Pathname Expansion above) to expand to a null string, rather than themselves.
```
For instance, if the runs dir is completely empty, this will not work correctly if the nullglob option is not set:
```
for RUN_DIR in ${IN_DIR}/*; do
    if [[ ! -d ${RUN_DIR} ]]; then
        continue
    fi
```

I think...